rust/emitter/src/emitter.rs

```diff
@@ -707,6 +712,24 @@ impl InstructionWriter {
         self.emit_op(Opcode::JumpTarget);
     }
 
+    pub fn bytecode_offset(&mut self) -> BytecodeOffset {
+        BytecodeOffset { offset: self.bytecode.len() }
+    }
+
+    pub fn patch_jump_target(&mut self, jumplist: Vec<BytecodeOffset>) {
+        let target_offset = self.bytecode.len();
```

Comment:
this is not the offset of jump target, but the opcode after the jump target.

I guess, it's better change this method to `emit_jump_target_and_patch`, and move `self.emit.jump_target();` after this line, so that the `target_offset` becomes the correct value and also the relation is clearer

Reply:
It used to be the offset of the jump target instruction, unless this got changed recently.
The reason being that the jump-target was used as the point where the code-coverage is attached to.

Reply:
Right, the jump must land on the JumpTarget instruction.